src/Education.UnitTests/Layout/TabGroupTests.cs

```diff
@@ -0,0 +1,102 @@
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Linq;
+using System.Windows.Controls;
+using System.Windows.Controls.Primitives;
+using System.Windows.Documents;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+using SonarLint.VisualStudio.Education.Layout;
+using SonarLint.VisualStudio.Education.Layout.Tabs;
+using TabItem = SonarLint.VisualStudio.Education.Layout.Tabs.TabItem;
+
+namespace SonarLint.VisualStudio.Education.UnitTests.Layout
+{
+    [TestClass]
+    public class TabGroupTests
+    {
+        [TestMethod]
+        public void CreateVisualization_GeneratesCorrectStructure()
+        {
+            var tabGroupName = "MainTabsGroup";
+            var tabNames = new[] { "why", "how_to_fix", "more_info" };
+            var tabDisplayNames = new[] { "Why is this an issue", "How to fix it?", "More info" };
+            var tabContents = new Block[]{ new Paragraph(), new Section(), new BlockUIContainer()};
+            var tabsRepositoryMock = new Mock<ITabsRepository>();
+
+            var testSubject = new TabGroup(tabGroupName,
+                tabNames.Select((name, i) => new TabItem(name, tabDisplayNames[i], SetUpTabMock(tabContents[i])))
+                    .ToList(), tabsRepositoryMock.Object);
+
+            var visualization = (Section)testSubject.CreateVisualization();
+
+            visualization.Blocks.FirstBlock.Should().BeOfType<BlockUIContainer>();
+            visualization.Blocks.Count.Should().Be(2);
+
+            VerifyButtonsGeneratedCorrectly(visualization, tabGroupName, tabNames, tabDisplayNames);
+            VerifyActiveTabIsPresent(visualization, tabGroupName, tabNames);
+            VerifyAllTabsAreRegistered(tabsRepositoryMock, tabGroupName, tabNames, tabContents);
+        }
+
+        private static void VerifyButtonsGeneratedCorrectly(Section visualizationRoot, string tabGroupName, string[] tabNames, string[] tabDisplayNames)
+        {
+            var buttonsBlock = (BlockUIContainer)visualizationRoot.Blocks.FirstBlock;
+            buttonsBlock.Name.Should().Be(tabGroupName);
+            buttonsBlock.Child.Should().BeOfType<StackPanel>();
+            var buttons = ((StackPanel)buttonsBlock.Child).Children.Cast<ToggleButton>().ToList();
+            buttons.Count.Should().Be(tabNames.Length);
+
+            for (var i = 0; i < buttons.Count; i++)
```

Comment:
instead maybe use Should().BeEquivalentTo()

Reply:
I think it's too complicated to do it like that. I would still need the for loop, but instead of checking the relevant properties inside the loop, I would need to put the elements in a list and then also set up BeEquivalentTo() to only check those 2 fields